tests/integration/test_distributeby.py

```diff
@@ -0,0 +1,33 @@
+import dask.dataframe as dd
+import pandas as pd
+import pytest
+
+try:
+    import cudf
+except ImportError:
+    cudf = None
+
+
+@pytest.mark.parametrize("gpu", [False, pytest.param(True, marks=pytest.mark.gpu)])
+def test_distribute_by(c, gpu):
+
+    if gpu:
+        xd = cudf
+    else:
+        xd = pd
+
+    df = xd.DataFrame({"id": [0, 1, 2, 1, 2, 3], "val": [0, 1, 2, 1, 2, 3]})
+    ddf = dd.from_pandas(df, npartitions=2)
+
+    c.create_table("test", ddf)
+    partitioned_ddf = c.sql(
+        """
+    SELECT
+    id
+    FROM test
+    DISTRIBUTE BY id
+    """
+    )
+    part_0_ids = partitioned_ddf.get_partition(0).compute().id.unique()
+    part_1_ids = partitioned_ddf.get_partition(1).compute().id.unique()
+    assert bool(set(part_0_ids) & set(part_1_ids)) is False
```

Comment:
This [fails in the GPU case]( https://gpuci.gpuopenanalytics.com/job/dask/job/dask-sql/job/prb/job/dask-sql-prb/151/CUDA_VER=11.2,LINUX_VER=ubuntu18.04,PYTHON_VER=3.8,RAPIDS_VER=21.12/testReport/tests.integration/test_distributeby/test_distribute_by_True_/ ) because neither of these are iterable

```python
>       assert bool(set(part_0_ids) & set(part_1_ids)) is False

tests/integration/test_distributeby.py:33: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/opt/conda/envs/dask_sql/lib/python3.8/site-packages/cudf/core/single_column_frame.py:92: in __iter__
    cudf.utils.utils.raise_iteration_error(obj=self)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

obj = 0    3
Name: id, dtype: int64

    def raise_iteration_error(obj):
>       raise TypeError(
            f"{obj.__class__.__name__} object is not iterable. "
            f"Consider using `.to_arrow()`, `.to_pandas()` or `.values_host` "
            f"if you wish to iterate over the values."
        )
E       TypeError: Series object is not iterable. Consider using `.to_arrow()`, `.to_pandas()` or `.values_host` if you wish to iterate over the values.
```

May need to coerce them to something that is iterable first. Maybe like this?

```suggestion
    if gpu:
        part_0_ids = part_0_ids.to_pandas()
        part_1_ids = part_1_ids.to_pandas()
    assert bool(set(part_0_ids) & set(part_1_ids)) is False
```